common/src/main/java/org/opensearch/ml/common/model/MLGuard.java

```diff
@@ -0,0 +1,133 @@
+        }
+    }
+
+    private void fillStopWordsToMap(@NonNull Guardrail guardrail, Map<String, List<String>> map) {
+        List<StopWords> stopWords = guardrail.getStopWords();
+        if (stopWords == null || stopWords.isEmpty()) {
+            return;
+        }
+        for (StopWords e : stopWords) {
+            map.put(e.getIndex(), Arrays.asList(e.getSourceFields()));
+        }
+    }
+
+    public Boolean validate(String input, int type) {
+        switch (type) {
+            case 0: // validate input
+                return validateRegexList(input, inputRegex) && validateStopWords(input, stopWordsIndicesInput);
+            case 1: // validate output
+                return validateRegexList(input, outputRegex) && validateStopWords(input, stopWordsIndicesOutput);
+            default:
+                return true;
+        }
+    }
+
+    public Boolean validateRegexList(String input, List<String> regexList) {
+        for (String regex : regexList) {
+            if (!validateRegex(input, regex)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public Boolean validateRegex(String input, String regex) {
+        Pattern pattern = Pattern.compile(regex);
+        Matcher matcher = pattern.matcher(input);
+        return !matcher.matches();
+    }
+
+    public Boolean validateStopWords(String input, Map<String, List<String>> stopWordsIndices) {
+        for (Map.Entry entry : stopWordsIndices.entrySet()) {
+            if (!validateStopWordsSingleIndex(input, (String) entry.getKey(), (List<String>) entry.getValue())) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public Boolean validateStopWordsSingleIndex(String input, String indexName, List<String> fieldNames) {
+        SearchRequest searchRequest;
+        SearchResponse searchResponse;
+        String queryBody;
+        Map<String, String> documentMap = new HashMap<>();
+        for (String field : fieldNames) {
+            documentMap.put(field, input);
+        }
+        Map<String, Object> queryBodyMap = Map
+                .of("query", Map.of("percolate", Map.of("field", "query", "document", documentMap)));
+
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            queryBody = AccessController.doPrivileged((PrivilegedExceptionAction<String>) () -> gson.toJson(queryBodyMap));
+            SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+            XContentParser queryParser = XContentType.JSON.xContent().createParser(xContentRegistry, LoggingDeprecationHandler.INSTANCE, queryBody);
+            searchSourceBuilder.parseXContent(queryParser);
+            searchSourceBuilder.size(1); //Only need 1 doc returned, if hit.
+            searchRequest = new SearchRequest().source(searchSourceBuilder).indices(indexName);
```

Comment:
SearchRequest return top 10 docs by default if I remember correctly. Should we increase the return doc size ? Should we add pagination to get all docs?